nixos/modules/services/misc/paperless.nix

```diff
@@ -326,18 +326,33 @@ in
       environment = env;
     };
 
-    systemd.services.paperless-web = {
+    systemd.services.paperless-web = let
+      secretKeyFile = "${cfg.dataDir}/nixos-paperless-secret-key";
+    in {
       description = "Paperless web server";
       # Bind to `paperless-scheduler` so that the web server never runs
       # during migrations
       bindsTo = [ "paperless-scheduler.service" ];
       after = [ "paperless-scheduler.service" ];
+      preStart = ''
+        if [ ! -f "${secretKeyFile}" ]; then
+          (
+            umask 0377
+            head /dev/urandom | tr -dc A-Za-z0-9 | head -c64 > "${secretKeyFile}"
```

Comment:
Now that I look at this again the first head should probably be a cat. The probability that this produces a string shorter than 64 characters should be very very low though.

Reply:
How about

```console
tr -dc A-Za-z0-9 </dev/urandom | head -c64 >"${secretKeyFile}"
```

Reply:
Right, the cat would be unnecessary as well